src/main/java/microservices/userServiceBasic/api/UserApiRest.java

```diff
@@ -0,0 +1,44 @@
+package microservices.userServiceBasic.api;
+
+import microservices.userServiceBasic.api.mapper.UserApiMapper;
+import microservices.userServiceBasic.api.model.UserDetails;
+import microservices.userServiceBasic.api.response.UserResponse;
+import microservices.userServiceBasic.domain.UserService;
+import microservices.userServiceBasic.domain.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class UserApiRest implements UserApi {
+
+	private final UserService userService;
+	private final UserApiMapper userApiMapper;
+
+	@Autowired
+	public UserApiRest(UserService userService, UserApiMapper userApiMapper) {
+
+		this.userService = userService;
+		this.userApiMapper = userApiMapper;
+	}
+
+	@Override
+	@RequestMapping(value = "/user", method = RequestMethod.POST, produces = "application/json")
+	public ResponseEntity<UserResponse> createUser(@RequestBody UserDetails userDetails) {
+
+		User user = userService.saveUser(userApiMapper.mapToUser(userDetails));
+		return ResponseEntity.ok(new UserResponse(userApiMapper.mapToUserDetails(user)));
+	}
+
+	@Override
+	@RequestMapping(value = "/user", method = RequestMethod.GET, produces = "application/json")
+	public ResponseEntity<UserResponse> getUser(@RequestBody UserDetails userDetails) {
```

Comment:
Usually GetUser should have some identifier as param, UUID or email. UserDetails contain exactly the info that you insert, so if I already have that info why should I query the service to get the same info? (apart from UUID). :) Let's do this Get from UUID so the method should be `getUser(@PathVariable UUID uuid)`, `@PathVariable` because you can query from the path and you should use `@RequestMapping(value = "/user/{uuid}"...`
take a look here: https://github.com/PauloFer1/pf-auth-server/blob/master/src/main/java/com/pfernand/pfauthserver/adapter/primary/api/AuthenticationApiController.java#L23